core/app/models/spree/promotion_code/code_builder.rb

```diff
@@ -26,7 +26,7 @@ def build_promotion_codes
   private
 
   def codes
-    if num_codes > 1
+    if (num_codes + promotion.codes.size) > 1
```

Comment:
I am also happy to rewrite the internal API to be more consistent. If I didn't care about backwards compatibility, I wouldn't have the CodeBuilder deciding whether to use a suffix based on how many codes there are at all, I'd have another option. If I did care about backwards compatibility... it may have to do some kind of `*args` checking thing. 

Initially I was going down that route, but I think the chances of me getting such a PR approved are not very high, so didn't want to do all that work only to have it sit here forever without getting approved.